environments/jvm/builder/Dockerfile

```diff
@@ -0,0 +1,47 @@
+## Fission builder base image
+ARG BUILDER_IMAGE=fission/builder:latest
+FROM ${BUILDER_IMAGE}
+
+## Section copied from the OpenJDK 8 Dockerfile
+
+ENV LANG C.UTF-8
+RUN { \
+    echo '#!/bin/sh'; \
+    echo 'set -e'; \
+    echo; \
+    echo 'dirname "$(dirname "$(readlink -f "$(which javac || which java)")")"'; \
+    } > /usr/local/bin/docker-java-home \
+    && chmod +x /usr/local/bin/docker-java-home
+ENV JAVA_HOME /usr/lib/jvm/java-1.8-openjdk
+ENV PATH $PATH:/usr/lib/jvm/java-1.8-openjdk/jre/bin:/usr/lib/jvm/java-1.8-openjdk/bin
+
+ENV JAVA_VERSION 8u171
+ENV JAVA_ALPINE_VERSION 8.171.11-r0
+
+RUN set -x \
+    && apk add --no-cache \
+    openjdk8="$JAVA_ALPINE_VERSION" \
+    && [ "$JAVA_HOME" = "$(docker-java-home)" ]
+
+## Section copied from the Maven Dockerfile
```

Comment:
We need to use `FROM:BUILDER` and then also add JDK as well as Maven. Since we can't polymorphically create images from 3 sources (BUILDER, JDK, Maven), this seemed like the best alternative. Are you aware of any better way that I can follow?

Reply:
Ah okay, I see. You could use a multistage build to copy the binaries and set the env vars yourself, but that doesn't reduce the dockerfile much. I am fine with keeping it like this. If you want to change anything, extract the Maven and Java versions to build args.